router/pkg/authentication/jwks_token_decoder.go

```diff
@@ -0,0 +1,43 @@
+package authentication
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/MicahParks/keyfunc/v2"
+	"github.com/golang-jwt/jwt/v5"
+)
+
+type tokenDecoder interface {
+	Decode(token string) (Claims, error)
+}
+
+type jwksTokenDecoder struct {
+	// JSON Web Key Set, automatically updated in the background
+	// by keyfunc.
+	jwks *keyfunc.JWKS
+}
+
+// Decode implements tokenDecoder.
+func (j *jwksTokenDecoder) Decode(tokenString string) (Claims, error) {
+	token, err := jwt.Parse(tokenString, j.jwks.Keyfunc)
+	if err != nil {
+		return nil, fmt.Errorf("could not validate token: %w", err)
+	}
+	claims := token.Claims.(jwt.MapClaims)
+	return Claims(claims), nil
+}
+
+func NewJwksTokenDecoder(url string, refreshInterval time.Duration) (tokenDecoder, error) {
+
+	jwks, err := keyfunc.Get(url, keyfunc.Options{
```

Comment:
This starts a goroutine which needs to be cleaned up `jwks.EndBackground()` across graph hot reloads and server shutdown. Implement a close() method.

Reply:
Can you give me a few more details regarding to where exactly the close() method should be called? I don't know exactly where the hot reloads or the server shutdown code is. 

Is The Close() method ok like this? 
https://github.com/wundergraph/cosmo/blob/a5653293018df745d9f9b99bf98f7b2d6fde9392/router/pkg/authentication/jwks_token_decoder.go#L45-L47

Reply:
The decoder is used router wide correct? In that case, you only need to call it [here](https://github.com/wundergraph/cosmo/blob/main/router/core/router.go#L1103).

Reply:
done